### PR TITLE
[FE] Add the send_email flag to the status of the incident

### PR DESCRIPTION
**Pull request by aditudorache**
This PR implements the possibility of sending an email for incidents that are not `deelmelding`.
Note that this depends of the API fucntionality SIG-2619 and won't work before this is released.

### CODE REVIEW

File: src/shared/types/index.js [hunk 1]
```diff
@@ -152,6 +152,7 @@ export const incidentType = PropTypes.shape({
   status: PropTypes.shape({
     state: PropTypes.string,
     state_display: PropTypes.string,
+    send_email: PropTypes.bool,
```
Comment (janjaap) [inline src/shared/types/index.js:155]: Do we know that `send_email` is the actual parameter that the API accepts/expects?

File: src/signals/incident-management/containers/IncidentDetail/components/StatusForm/components/CheckboxInput/index.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+const CheckboxInput = ({ name, label }) => ({ handler }) => (
+  <CheckboxStyle>
+    <Checkbox id={name} name={name} checked={handler().value} onChange={handler().onChange} />
+    <Label htmlFor={name} label={label} />
+  </CheckboxStyle>
+);
+
+CheckboxInput.propTypes = {
+  name: PropTypes.string,
+  label: PropTypes.string,
```
Comment (janjaap) [inline src/signals/incident-management/containers/IncidentDetail/components/StatusForm/components/CheckboxInput/index.js:25]: Shouldn't both props be required?

File: src/signals/incident-management/containers/IncidentDetail/components/StatusForm/index.js [hunk 3]
```diff
@@ -115,7 +123,7 @@ const StatusForm = ({ defaultTexts, incident, onClose }) => {
 
       form.controls.text.updateValueAndValidity();
     });
-  }, [form.controls.status.valueChanges, form.controls.text]);
+  }, [form.controls.status.valueChanges, form.controls.text, form.controls.send_email.valueChanges]);
```
Comment (janjaap) [inline src/signals/incident-management/containers/IncidentDetail/components/StatusForm/index.js:126]: I don't see `form.controls.send_email.valueChanges` being used in the `useEffect` handler

File: src/signals/incident-management/containers/IncidentDetail/components/StatusForm/index.js [hunk 4]
```diff
@@ -188,6 +196,20 @@ const StatusForm = ({ defaultTexts, incident, onClose }) => {
...
+            </Notification>
+
+            {!isDeelmelding && (
+              <FieldControlWrapper
+                control={form.get('send_email')}
+                data-testid="statusFormSendEmailField"
+                name="send_email"
+                label={MELDING_CHECKBOX_DESCRIPTION}
+                render={CheckboxInput}
+              />
```
Comment (janjaap) [inline src/signals/incident-management/containers/IncidentDetail/components/StatusForm/index.js:210]: There doesn't seem to be a test for this element

File: src/shared/types/index.js (hunk 1 above)
Comment (aditudorache) [reply src/shared/types/index.js:155]: Yes, I checked that with the backend-ers

File: src/signals/incident-management/containers/IncidentDetail/components/StatusForm/index.js (hunk 3 above)
Comment (aditudorache) [reply src/signals/incident-management/containers/IncidentDetail/components/StatusForm/index.js:126]: Correct, I removed it!

File: src/signals/incident-management/containers/IncidentDetail/components/StatusForm/components/CheckboxInput/index.js (hunk 2 above)
Comment (aditudorache) [reply src/signals/incident-management/containers/IncidentDetail/components/StatusForm/components/CheckboxInput/index.js:25]: Yes they should! :-)

File: src/signals/incident-management/containers/IncidentDetail/components/StatusForm/index.js (hunk 4 above)
Comment (aditudorache) [reply src/signals/incident-management/containers/IncidentDetail/components/StatusForm/index.js:210]: done

**Comment by janjaap** [comment]
@aditudorache Tests are failing

**Pull request merged**